games/api/serializers.py

```diff
@@ -64,6 +65,11 @@ def get_time_start(self, game):
     def get_time_end(self, game):
         return game.time_end.strftime("%H:%M") if game.time_end else ""
 
+    def get_report_notes(self, game):
+        if self.context["request"].user.profile != game.dm:
```

Comment:
Actually it should be visible to ANY DM in the system, so the role of the profile should be checked. AFAIR there's a method/property for that.